RxSwift/Disposables/DisposeBase.swift

```diff
@@ -8,7 +8,7 @@
 
 /// Base class for all disposables.
 public class DisposeBase {
-    init() {
+    public init() {
```

Comment:
Required as there are various classes that derive from DisposeBase that have a public init method, and the generated .swiftinterface files don't like this.
Seems like it probably wants to be public anyway as the class is public

Reply:
Again, really preferable to avoid this. There is no value in init'ing this directly.

Reply:
ok. I'll see if I can come up with some other change which leaves this internal but stops it getting emitted into the .swiftinterface. It feels like this might be a bug in swift, but not sure yet

Reply:
FYI I couldn't infer any way work around this, I filed a bug on swift:
https://bugs.swift.org/browse/SR-13108

See what they say about it

Reply:
@freak4pc Not sure if you saw the commentary from swift/apple over on their JIRA - it looks like this is a bug in the swift compiler and may be related to having "enable testability" set on the RxSwift project. We should not need to change the RxSwift source code here.

It seemed like a good solution would be to disable testability for release builds (as Apple suggest, this flag alters the meaning of public/private and is undesirable for release builds anyway). I went to validate this and found however that even without the testability flag, the bug still occurs and the swiftinterface file is still broken. I'm waiting now for apple to confirm that I've got my flags right.

If disabling testability proves not to work, what comes to my mind next is a post-build step which patches the generated swiftinterface files to work around the bug.
It's a hack, but I'll probably do this internally for my company's own copies of the RxSwift binary framework. How do you feel about the idea of that for RxSwift proper though?

Reply:
@borland Yeah I commented below I don't mind doing disable testability at all, this should be fine. It's unfortunate this doesn't solve it ...

A post-build step is not a great step in the right direction IMO and I'm worried it could cause issues for other dependency managers, or even worse for folks using submodules.

Lets keep our fingers crossed around these, seems like Robert is actually quite responsive given he already had a PR around the testability thing.

BTW did you consider removing the testability flag altogether just to test if it makes any change?

Reply:
I set the “enable testability” flag to No for all configurations in Xcode, and it was still bugged. I’m not 100% sure if that’s the right flag so waiting to see what Robert says. 
Agree a post build step is not good

Reply:
re-reviewed against the latest commits